tests/test-utils.js

```diff
@@ -0,0 +1,21 @@
+class TestUtils {
+  static async waitExits (pool, numberOfExitEventsToWait) {
+    let exitEvents = 0
+    return new Promise(function (resolve, reject) {
+      pool.workers.forEach(w => {
+        w.on('exit', () => {
```

Comment:
If we can also track the worker creation event in a pool, we should be able to get rid of sleep() usage.

Reply:
Unfortunately is not exactly true....where we now use setTimeout is just because we are not expecting threads to exit.

A worker created reflects immediately on the workers size that we can check and for this reason is not problematic.

Reply:
@jerome-benoit  I guess you agree to close this conversation, right ?